packages/extensions/modelerfour/src/modeler/modelerfour.ts

```diff
@@ -2146,8 +2146,27 @@ export class ModelerFour {
   processResponses(httpOperation: OpenAPI.HttpOperation, operation: Operation) {
     const acceptTypes = new Set<string>();
 
+    const modelerResponses = httpOperation.responses;
+
+    // If the operation has final-state-schema lro option, add a "200" response if there isn't one already
+    if (httpOperation["x-ms-long-running-operation-options"]?.["final-state-schema"]) {
+      const finalStateSchema = httpOperation["x-ms-long-running-operation-options"]?.["final-state-schema"] as string;
+      if (!Object.keys(modelerResponses).some((k) => k === "200")) {
+        modelerResponses["200"] = {
```

Comment:
Its possible a 200 is defined but without a response body - in which case we would want to use the final-state-schema?

For long running PUTs, 200 and 201 are most likely always expected but it would be desirable to have the final schema specified through final-state-schema?

Reply:
> Its possible a 200 is defined but without a response body - in which case we would want to use the final-state-schema?

I think it's unclear what the intent is here so I'd be reluctant to modify an existing 200 response. If the intent is to say that the operation _could_ return a 200 with no body then adding the final response schema is wrong. But if the operation does not actually return 200, then the right fix is to remove the 200 response and then the final-state-schema will work as intended.

> For long running PUTs, 200 and 201 are most likely always expected but it would be desirable to have the final schema specified through final-state-schema?

Long-running PUTs should return a 200 or 201 _AND_ a response body, and I think this should be done with standard OpenAPI. No extension needed.

> The other option would be that we never allow a 200 without a response schema defined. Not sure if there is a linter rule for this.

We do flag this with the data-plane linter.

Reply:
Opened https://github.com/Azure/cadl-azure/issues/2266 to track.